src/core/SP_reader.sol

```diff
@@ -0,0 +1,217 @@
+     * @param _veraxAddress The new address of the Verax Registry.
+     */
+    function updateVeraxAddress(address _veraxAddress) external onlyOwner {
+        ReaderStorage storage $ = _getReaderStorage();
+        if (_veraxAddress == address(0)) revert Invalid();
+        $.veraxRegistry = IVerax(_veraxAddress);
+    }
+
+    /**
+     * @notice Updates the address of the Sign Protocol Contract.
+     * @dev Only the owner of the contract can call this function. Reverts if the provided address is the zero address.
+     * @param _spAddress The new address of the Sign Protocol Contract.
+     */
+    function updateSPAddress(address _spAddress) external onlyOwner {
+        ReaderStorage storage $ = _getReaderStorage();
+        if (_spAddress == address(0)) revert Invalid();
+        $.signProtocol = SP(_spAddress);
+    }
+
+    /**
+     * @param _id The unique id of the attestation. Takes a bytes32 as this is the datatype Verax uses
+     * @return Sign Protocol Attestation
+     */
+    function getAttestationMethod1(bytes32 _id) external view returns (SPAttestation memory) {
+        ReaderStorage storage $ = _getReaderStorage();
+
+        uint256 spId = uint256(_id); // Converting bytes32 to uint256 for comparison
+
+        if (spId > type(uint64).max) {
+            // ID is out of range for Sign Protocol
+            VeraxAttestation memory vAttestation = $.veraxRegistry.getAttestation(_id);
+            return _convertVeraxToSP_nullID(vAttestation); // Attestation ID is out of range for Sign Protocol
+        } else {
+            // ID is in the range for Sign Protocol
+            SPAttestation memory attestation = $.signProtocol.getAttestation(uint64(spId)); // uint256 spID -> uint64
+            if (attestation.attestTimestamp != 0) {
+                // Checks if attestation is empty
+                return attestation;
+            } else {
+                // If SP Protocol Attestation does not exist
+                VeraxAttestation memory vAttestation = $.veraxRegistry.getAttestation(_id);
+                return _convertVeraxToSP(vAttestation, uint64(spId));
+            }
+        }
+    }
+
+    /**
+     * @notice Helper Function: Transforms a Verax Attestation to a Sign Protocol Attestation
+     * @param _vAttestation - The Verax Attestation
+     * @param _id - AttestationId
+     * @return Sign Protocol Attestation
+     */
+    function _convertVeraxToSP(
+        VeraxAttestation memory _vAttestation,
+        uint64 _id
+    )
+        private
+        pure
+        returns (SPAttestation memory)
+    {
+        SPAttestation memory attestation;
+
+        if (uint256(_vAttestation.schemaId) > type(uint64).max) {
+            attestation.schemaId = 0; // Leaving it null
+        } else {
+            attestation.schemaId = uint64(uint256(_vAttestation.schemaId)); // bytes32 -> uint256 -> uint64
+        }
+
+        attestation.linkedAttestationId = _id;
```

Comment:
Not sure if this is correct, the `Attestation` struct in SP does not store the ID.